assets/src/edit-story/components/panels/design/textStyle/textStyle.js

```diff
@@ -30,16 +30,14 @@ import StyleControls from './style';
 import ColorControls from './color';
 import FontControls from './font';
 
-function StylePanel(props) {
-  const fontDropdownRef = useRef(null);
-  const textColorRef = useRef(null);
+const StylePanel = forwardRef(function StylePanel(props, ref) {
```

Comment:
What's the reason for the change to use `forwardRef` in this component here?

Reply:
```
  console.error
    Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

    Check the render method of `FontPicker`.
        at mockConstructor (/Users/jonny/workspace/web-stories-wp/node_modules/jest-mock/build/index.js:171:19)
        at FontPicker (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/fontPicker.js:38:3)
        at div
        at row__Row (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
        at FontControls (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/font.js:64:3)
        at div
        at content__Container (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
        at Content (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/panel/shared/content.js:40:20)
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at section
        at panel__Wrapper (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
        at Panel (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/panel/panel.js:45:3)
        at SimplePanel (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/panel/simplePanel.js:29:24)
        at textStyle___StyledSimplePanel (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
        at StylePanel (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/textStyle.js:34:27)
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at Wrapper (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/test/textStyle.js:57:20)
        at TestPanel (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/shared/test/_utils.js:34:3)
        at Object.<anonymous>.exports.ThemeProvider (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/ServerStyleSheet.js:66:13)

      at printWarning (node_modules/react-dom/cjs/react-dom.development.js:67:30)
      at error (node_modules/react-dom/cjs/react-dom.development.js:43:5)
      at validateFunctionComponentInDev (node_modules/react-dom/cjs/react-dom.development.js:17926:9)
      at mountIndeterminateComponent (node_modules/react-dom/cjs/react-dom.development.js:17893:7)
      at beginWork (node_modules/react-dom/cjs/react-dom.development.js:19049:16)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:23940:14)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:22779:12)

  console.error
    Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

    Check the render method of `ColorControls`.
        at mockConstructor (/Users/jonny/workspace/web-stories-wp/node_modules/jest-mock/build/index.js:171:19)
        at div
        at row__Row (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
        at ColorControls (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/color.js:29:26)
        at div
        at content__Container (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
        at Content (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/panel/shared/content.js:40:20)
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at section
        at panel__Wrapper (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
        at Panel (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/panel/panel.js:45:3)
        at SimplePanel (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/panel/simplePanel.js:29:24)
        at textStyle___StyledSimplePanel (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
        at StylePanel (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/textStyle.js:34:27)
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at /Users/jonny/workspace/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
        at Wrapper (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/test/textStyle.js:57:20)
        at TestPanel (/Users/jonny/workspace/web-stories-wp/assets/src/edit-story/components/panels/shared/test/_utils.js:34:3)
        at Object.<anonymous>.exports.ThemeProvider (/Users/jonny/workspace/web-stories-wp/node_modules/styled-components/src/models/ServerStyleSheet.js:66:13)

      at printWarning (node_modules/react-dom/cjs/react-dom.development.js:67:30)
      at error (node_modules/react-dom/cjs/react-dom.development.js:43:5)
      at validateFunctionComponentInDev (node_modules/react-dom/cjs/react-dom.development.js:17926:9)
      at mountIndeterminateComponent (node_modules/react-dom/cjs/react-dom.development.js:17893:7)
      at beginWork (node_modules/react-dom/cjs/react-dom.development.js:19049:16)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:23940:14)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:22779:12)
```

I got the following javascript unit test error. I believe it is unrelated, but it did  block this PR. It is  worth note, this warning was in other test runs, but for some reason it change to a error here.

Reply:
There's a dedicated ticket for this: #7875

cc @miina

Reply:
I believe that @miina is away ATM. 

The difference is, some reason, it become an error over a warning here. @swissspidy should I skip the test for now?

Reply:
Let's fix #7875 first in a separate PR. It might be fixed by the change you did here, but I'd want this to be in a dedicated PR & reviewed by @miina.

Reply:
Broken the fix into it's own PR - https://github.com/google/web-stories-wp/pull/8100. 

I have skipped the test for now.

Reply:
I'd rather want to fix the actual bug instead of just skipping tests and hiding potential problems.

Reply:
> I'd rather want to fix the actual bug instead of just skipping tests and hiding potential problems

Can we not fix it in #7875 ? I personally tried to fix it and I don't know how.

Reply:
i put this on our plate for the next sprint, we just are swamped this week trying to finish our work for this upcoming release.

Reply:
Alright then. I'm still not very comfortable just skipping tests ignoring potential bugs, but if I know it will get addressed soon then I suppose it should be ok.